streams/src/test/java/org/apache/kafka/streams/integration/GlobalKTableIntegrationTest.java

```diff
@@ -337,7 +341,18 @@ public void shouldGetToRunningWithOnlyGlobalTopology() throws Exception {
         startStreams();
         waitForApplicationState(singletonList(kafkaStreams), State.RUNNING, Duration.ofSeconds(30));
 
-        kafkaStreams.close();
```

Comment:
I am personally still not sure if we actually want/need this change in the test...

It's also somewhat incomplete, as it does not cover the initialization phase, not do we actually process data in this test and thus don't really hit the actual runtime path.

Last, if we test for the absence of the log of a clean shutdown, it seem we should also test for their presents in case of an error. But testing logging seems to get out of hand this way.

Overall, I would still be a favor to not apply this change to the tests as it seems pointless (and potentially confusing).

Reply:
I somehow agree. But it also seems like we are missing tests for the error case where the state of the Streams client closes to `ERROR`. I am fine with removing the change of the test from this PR and just make a one-time manual verification of the existence and absence of the log message. However, then we need a separate PR for the missing test or a ticket that documents that a test is missing.
Have you had a look into `GlobalStreamThreadTest`? Could we add something there?

Reply:
>But it also seems like we are missing tests for the error case where the state of the Streams client closes to ERROR.

Not sure what this test should verify exactly? The final state of the client? Some log messages?

> However, then we need a separate PR for the missing test or a ticket that documents that a test is missing.

Happy to do a follow up PR.

> Have you had a look into GlobalStreamThreadTest? Could we add something there?

I don't think that `GlobalStreamThreadTest` works, because it tests the thread in isolation, while we want to test the interaction with `KafkaStream` client? (At least from my current understanding from you comment above). Maybe we could add a test to `KafkaStreamsTest` though?

Let me update this PR removing this test change so we can merge it, and work on a follow up PR afterwards.